### DISCUSSION TITLE
Export json of settings

### DISCUSSION THREAD

**Discussion by zoomyboy** [tailwindlabs/tailwindcss#2072] [Ideas]
I would like Tailwind to export a json file of the settings when they're compiled down as a big object and before the actual CSS is generated (meaning when all the custom plugins are applied and the default config is merged with the ``extend { ... }`` custom settings.

I am building a page builder in the backend of a CMS (which is a modification of the Colorpicker that October CMS uses) where it would be nice to select any available tailwind color from the backend so that the system knows which corresponding class to use (e.g. ``bg-blue-500``) to apply.

For this to work smoothly, that information should someshow be available to any other 3rd party service who wants to interact with the current settings (because you could present the visual color to the user). I am thinking of an ``'export'`` setting inside of the ``corePlugins`` section where you could define wether that particular feature should be written to its own file.

That could probably be created by a utility plugin as well where you have access to the config vars, but that seems rather hacky, because

1. A Plugin is used to insert new features into the configuration rather than extracting values out of it
2. You would have to install that (lets say it's a plugin) on every new site who uses that functionality.

An export for colors could look like this: 

```
{
    "bg-blue-100": "#...",
    "bg-blue-200": "#..."
...
}
```

Let me know your thoughts about this 😄

**Comment by royvanv** [comment]
I think you might want to look into `resolveConfig`:
https://tailwindcss.com/docs/configuration/#referencing-in-javascript

**Comment by zoomyboy** [comment]
I actually found a laravel mix plugin that does exactly that.

https://github.com/tylerwiegand/laravel-mix-tailwind-extract
